starboard/extension/time_zone.h

```diff
@@ -0,0 +1,42 @@
+// Copyright 2023 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef STARBOARD_EXTENSION_TIME_ZONE_H_
+#define STARBOARD_EXTENSION_TIME_ZONE_H_
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define kStarboardExtensionTimeZoneName "dev.starboard.extension.TimeZone"
+
+typedef struct StarboardExtensionTimeZoneApi {
+  // Name should be the string |kStarboardExtensionSetTimeZoneName|.
+  // This helps to validate that the extension API is correct.
+  const char* name;
+
+  // This specifies the version of the API that is implemented.
+  uint32_t version;
+
+  void (*SetTimeZone)(const char* time_zone_name);
```

Comment:
Add a brief description what the function does? 
Also let's specify that it should not be called with NULL or empty string.

I think a bool return value might be helpful too. Not sure how it would be handle if it fails to set, but silently failing is worse.